File: mvn-advent-of-code/src/test/java/advent/twenty_fifteen/Day13KnightsOfTheDinnerTableTest.java
package advent.twenty_fifteen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.*;

public class Day13KnightsOfTheDinnerTableTest {
    @Test
    public void checkReal() {
        DailyProblem<Integer, Integer> dp = new Day13KnightsOfTheDinnerTable("/twenty_fifteen/Day13-KnightsOfTheDinnerTable-input.txt");

        assertEquals(664, dp.getPart1Answer().intValue());
        assertEquals(640, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<Integer, Integer> dp = new Day13KnightsOfTheDinnerTable("/twenty_fifteen/Day13-KnightsOfTheDinnerTable-example#1.txt");

        assertEquals(330, dp.getPart1Answer().intValue());
        assertEquals(286, dp.getPart2Answer().intValue());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_fifteen/Day13KnightsOfTheDinnerTableTest.scala
package advent.twenty_fifteen

import org.scalatest.{FlatSpec, Matchers}

class Day13KnightsOfTheDinnerTableTest extends FlatSpec with Matchers {
  "2015 Day 13 - Input" should "calculate answers" in {
    val dp = new Day13KnightsOfTheDinnerTable("/twenty_fifteen/Day13-KnightsOfTheDinnerTable-input.txt")

    dp.part1Answer should be(664)
    dp.part2Answer should be(640)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day13KnightsOfTheDinnerTable("/twenty_fifteen/Day13-KnightsOfTheDinnerTable-example#1.txt")

    dp.part1Answer should be(330)
    dp.part2Answer should be(286)
  }
}
